Backend/dev/config-sync/server.py

```
@@ -205,6 +214,64 @@ def _patch_argv(body):
+    already allows anonymous reads from the VPN (see AllowListFromVPN /
+    AllowGetPutFromVPN), same as fetch_patch_zip_from_cloudfront() does for
+    the zip itself. 404 just means nobody's published this direction yet —
+    not an error, just an empty list."""
+    url = f"{_direction_base_url(direction)}/metadata.json"
+    try:
+        with urllib.request.urlopen(url, timeout=15) as resp:
+            data = json.loads(resp.read())
+    except urllib.error.HTTPError as e:
+        if e.code == 404:
+            return []
+        raise RuntimeError(f"fetching {url}: HTTP {e.code}")
```

Ruff (0.16.3)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)